### ISSUE TITLE
Add tanstack query.

### ISSUE THREAD

**Issue by Eduard-Hasa** [shadcn/taxonomy#46]
This project is really nice. Ideally any production project is going to need a library like tanstack/react query. Would be really nice if you incorporated it into this project

**Comment by shadcn** [comment]
> Ideally any production project is going to need a library like tanstack/react query.

Yep. I've yet to work on a Next.js app where I did not use react-query.

However, the reason I've not added it (yet) is, the Next team is currently looking into new ways for fetching data, mutating, caching and invalidating in server components.

I want to see how far we can go with just Next before we need react-query.

I'll come back to this when we do.

**Comment by Eduard-Hasa** [comment]
@shadcn im surprised I dont see custom hooks wrapping tanstack hooks more often to add type safety. I got a pretty cool setup where im using tanstack query at introduced type safety manually instead of using something like trpc. Would love to chat about your thoughts on the best approach.

**Comment by fergy-hs** [comment]
I would love to see the option to use RQ in this project. I have seen some examples around that show how to set up RQ to work with SSR but I think that at the current state it's just an overhead.

**Comment by fergiar** [comment]
I would love to see the option to use RQ in this project. I have seen some examples around that show how to set up RQ to work with SSR but I think that at the current state it's just an overhead.
